modules/uploads/scripts/prepare.ts

```diff
@@ -0,0 +1,134 @@
+import { RuntimeError, ScriptContext } from "../_gen/scripts/prepare.ts";
+import {
+	getMaxBytes,
+	PresignedUpload,
+	prismaToOutput,
+	UploadFile,
+} from "../utils/types.ts";
+import { getPresignedPutUrl } from "../utils/bucket.ts";
+
+export interface Request {
+	userId?: string;
+	files: UploadFile[];
+}
+
+export interface Response {
+	upload: PresignedUpload;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+	// If there is a userId, ensure the user exists
+	if (req.userId) {
+		await ctx.modules.users.getUser({
+			userIds: [req.userId],
+		});
+	}
+
+	// Ensure there are files in the upload
+	if (req.files.length === 0) {
+		throw new RuntimeError("no_files");
+	}
+
+	// Ensure the number of files is within the limit
+	if (req.files.length > ctx.userConfig.maxFilesPerUpload) {
+		throw new RuntimeError(
+			"too_many_files",
+			{
+				cause: `Max files per upload is ${ctx.userConfig.maxFilesPerUpload}` +
+					` (requested upload of ${req.files.length} files)`,
```

Comment:
Would be better to just include files len and max files per upload as meta instead of a full message. That's what the error description is for. Cause is for if there was a different error that caused this error, not the message.